python/pyspark/sql/dataframe.py

```diff
@@ -400,10 +402,19 @@ def toLocalIterator(self):
 
         >>> list(df.toLocalIterator())
         [Row(age=2, name=u'Alice'), Row(age=5, name=u'Bob')]
+        >>> it = df.toLocalIterator()
+        >>> import time
+        >>> time.sleep(5)
+        >>> next(it)
```

Comment:
It might make sense to test consuming the second element given that the fix eagerly serves the first element, (and this additional test probably belongs in tests.py rather than a doctest though).